### PR TITLE
Add a table block and small tweak to report section headings

### PR DESCRIPTION
**Pull request by helenb**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1440614544)

There is no design but Jeth has seen a demo already.

### Description of Changes Made
- Adds a typed table block that allows rich text
- Can be added on any page apart from the home page
- Adds it to the pattern library, including on various custom page templates and the styleguide
- Makes a small tweak to report section headings, spotted by Jeth while testing - ensures there is always space between the heading on the left and the content on the right

### How to Test

On a local build you can test in the pattern library

http://localhost:8000/pattern-library/pattern/patterns/molecules/streamfield/blocks/typed_table_block.html
http://localhost:8000/pattern-library/pattern/patterns/styleguide/components/components.html
http://localhost:8000/pattern-library/pattern/patterns/pages/blog/blog_detail.html

### Screenshots

<details>
  <summary>Expand to see more</summary>

Desktop

Screenshot 2024-06-04 at 10 12 03

Mobile, with hint to scroll

Screenshot 2024-06-04 at 10 13 07

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Updated field spec (https://docs.google.com/spreadsheets/d/1v1yqcSC54Vag4mxmTp8e6ZrnxITqzbjCA4A9XPQdXk8/edit?usp=sharing)
- [ ] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [ ] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [x] Automated WCAG 2.1 tests pass
- [x] HTML validation passes
- [x] Manual WCAG 2.1 tests completed
- [x] I have tested in a screen reader
- [x] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [x] Not required

#### Pattern library

- [x] The pattern library component for this template displays correctly, and does not break parent templates
- [x] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/static_src/javascript/components/table-hint.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            if (this.node.scrollLeft > 0) {
+                this.node.classList.add('is-scrolling');
+            }
+        });
+
+        this.button.addEventListener('click', () => {
+            this.node.scroll({
+                top: 0,
+                left: 500,
+                behavior: 'smooth',
```
Comment (chris-lawton) [inline tbx/static_src/javascript/components/table-hint.js:29]: Q: Does this need an alternative if the user has a reduced motion preference?

**Comment by chris-lawton** [comment]
This looks great @helenb - nice one. I thought i'd spotted an issue with the table border in high contrast mode but it seems that it might be a bug in dev tools as I had to refresh when changing the emulation type.

<details><summary>Video</summary>
<p>


https://github.com/torchbox/torchbox.com/assets/31627284/f4aae248-aa0d-4f58-9b20-ae74718507b4



</p>
</details>

**Review by chris-lawton — APPROVED** [review]

File: tbx/core/blocks.py [hunk 2]
```diff
@@ -740,6 +743,23 @@ class Meta:
         template = "patterns/molecules/streamfield/blocks/embed_block.html"
 
 
+class TypedTableBlock(blocks.StructBlock):
+    table = WagtailTypedTableBlock(
```
Comment (chris-lawton) [inline tbx/core/blocks.py:747]: WK has a line for `caption = blocks.CharBlock(required=False)` here - it doesn't look like we need that as I can see the caption in the Wagtail.

File: tbx/static_src/sass/components/_table.scss [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        height: 100%;
+        justify-content: flex-end;
+        align-items: center;
+        background: linear-gradient(
+            to right,
+            var(--color--background--overlay) 0%,
+            var(--color--background) 100%
+        );
+        opacity: 1;
+        transition: opacity 0.25s ease;
```
Comment (chris-lawton) [inline tbx/static_src/sass/components/_table.scss:62]: Could be `transition: opacity $transition;` if you were happy for the css to be `transition: opacity 0.25s ease-out;`

**Comment by helenb** [comment]
> This looks great @helenb - nice one. I thought i'd spotted an issue with the table border in high contrast mode but it seems that it might be a bug in dev tools as I had to refresh when changing the emulation type.
> 
> Video
>  Screen.Recording.2024-06-04.at.15.48.26.mov

I'm also seeing this when testing prefers-reduced-motion this morning - it isn't clearing, so think it is a bug.

File: tbx/core/blocks.py (hunk 2 above)
Comment (helenb) [reply tbx/core/blocks.py:747]: I think I need to update kit - possibly with a wagtail upgrade the type table block has changed - I started with a caption field and found I had it twice.

File: tbx/static_src/javascript/components/table-hint.js (hunk 1 above)
Comment (helenb) [reply tbx/static_src/javascript/components/table-hint.js:29]: good spot - will change that, and the transition in the css

**Pull request merged**